polkadot/node/network/statement-distribution/src/v2/mod.rs

```diff
@@ -3333,12 +3295,16 @@ pub(crate) fn answer_request(state: &mut State, message: ResponderMessage) {
 		validated_in_group: !mask.validated_in_group.clone(),
 	};
 
-	// Ignore disabled validators from the latest state when sending the response.
-	let disabled_mask =
-		per_session.disabled_bitmask(group_index).expect("group existence checked; qed");
+	// Ignore disabled validators when sending the response.
+	let disabled_mask = relay_parent_state.disabled_bitmask(group);
```

Comment:
not sure if this is needed anymore

Reply:
Why not?

Correct me if I'm wrong but that means that a validator requesting a candidate within a context of a relay parent in which he is disabled will simply be ignored. Seems like a good property

Reply:
Discussed in sync - checks are already made when we receive requests. It was originally there due to raciness issues (consequence of using the leafs). Raciness should no longer be there with relay parent perspective so this extra check is unnecessary.